modules/rostests/apitests/advapi32/RegCreateKeyExW.c

```diff
@@ -0,0 +1,158 @@
+#define _INC_WINDOWS
+#define COM_NO_WINDOWS_H
+#include <windef.h>
+#include <aclapi.h>
+#include <tchar.h>
+#include <debug.h>
+
+
+START_TEST(RegCreateKeyExW)
+{
+    HKEY hkey_main;
+    DWORD dwRes, dwDisposition;
+    PACL pACL = NULL;
+    PSECURITY_DESCRIPTOR pSD = NULL;
+    PSID pEveryoneSID = NULL, pAdminSID = NULL;
+    SID_IDENTIFIER_AUTHORITY SIDAuthWorld = {SECURITY_WORLD_SID_AUTHORITY};
+    SID_IDENTIFIER_AUTHORITY SIDAuthNT = {SECURITY_NT_AUTHORITY};
+    EXPLICIT_ACCESS ea[2];
+    SECURITY_ATTRIBUTES sa;
+    LONG lRes;
+    BOOL bRes;
+    LONG ErrorCode = 0;
+    HKEY hkSub = NULL;
+
+    /* If any of the test keys already exist, delete them to ensure proper testing*/
+    if (RegOpenKeyExW(HKEY_CURRENT_USER, L"mykey", 0, KEY_ALL_ACCESS, &hkey_main) == ERROR_SUCCESS)
+        RegDeleteKeyExW(HKEY_CURRENT_USER, L"mykey", KEY_ALL_ACCESS, 0);
+
+    if (RegOpenKeyExW(HKEY_CURRENT_USER, L"mykey1", 0, KEY_ALL_ACCESS, &hkey_main) == ERROR_SUCCESS)
+        RegDeleteKeyExW(HKEY_CURRENT_USER, L"mykey1", KEY_ALL_ACCESS, 0);
+
+    if (RegOpenKeyExW(HKEY_CURRENT_USER, L"mykey2", 0, KEY_ALL_ACCESS, &hkey_main) == ERROR_SUCCESS)
+        RegDeleteKeyExW(HKEY_CURRENT_USER, L"mykey2", KEY_ALL_ACCESS, 0);
+
+    /* Setup GetLastError to known value for tests */
+    SetLastError(0xdeadbeef);
+
+    // Create a well-known SID for the Everyone group.
+    bRes = AllocateAndInitializeSid(&SIDAuthWorld, 1,
+        SECURITY_WORLD_RID,
+        0, 0, 0, 0, 0, 0, 0,
+        &pEveryoneSID);
+    ok(bRes == TRUE, "AllocateAndInitializeSid Error %ld\n", GetLastError());
+    if (bRes != TRUE) 
+        goto Cleanup;
+
+    // Initialize an EXPLICIT_ACCESS structure for an ACE.
+    // The ACE will allow Everyone read access to the key.
+    ZeroMemory(&ea, 2 * sizeof(EXPLICIT_ACCESS));
+    ea[0].grfAccessPermissions = KEY_READ;
+    ea[0].grfAccessMode = SET_ACCESS;
+    ea[0].grfInheritance= NO_INHERITANCE;
+    ea[0].Trustee.TrusteeForm = TRUSTEE_IS_SID;
+    ea[0].Trustee.TrusteeType = TRUSTEE_IS_WELL_KNOWN_GROUP;
+    ea[0].Trustee.ptstrName  = (LPTSTR) pEveryoneSID;
+
+    // Create a SID for the BUILTIN\Administrators group.
+    bRes = AllocateAndInitializeSid(&SIDAuthNT, 2,
+        SECURITY_BUILTIN_DOMAIN_RID,
+        DOMAIN_ALIAS_RID_ADMINS,
+        0, 0, 0, 0, 0, 0,
+        &pAdminSID);
+    ok(bRes == TRUE, "AllocateAndInitializeSid Error %ld\n", GetLastError());
+    if (bRes != TRUE) 
+        goto Cleanup;
+
+    // Initialize an EXPLICIT_ACCESS structure for an ACE.
+    // The ACE will allow the Administrators group full access to the key.
+    ea[1].grfAccessPermissions = KEY_ALL_ACCESS;
+    ea[1].grfAccessMode = SET_ACCESS;
+    ea[1].grfInheritance= NO_INHERITANCE;
+    ea[1].Trustee.TrusteeForm = TRUSTEE_IS_SID;
+    ea[1].Trustee.TrusteeType = TRUSTEE_IS_GROUP;
+    ea[1].Trustee.ptstrName  = (LPTSTR) pAdminSID;
+
+    // Create a new ACL that contains the new ACEs.
+    dwRes = SetEntriesInAcl(2, ea, NULL, &pACL);
```

Comment:
```suggestion
    dwRes = SetEntriesInAclW(_countof(ea), ea, NULL, &pACL);
```
Needs the explicit unicode version if you want to implement ThFabba's suggestion.

Reply:
Done. Thanks.